src/components/notifications/Notifications.tsx

```diff
@@ -0,0 +1,286 @@
+            ],
+            { useNativeDriver: false }
+          )(e, { moveX: Math.min(gestureState.moveX / width, MAX_BG_OPACITY) })
+        }
+      } else {
+        if (gestureState.dx > 0) {
+          Animated.event(
+            [
+              null,
+              {
+                dx: translationAnim
+              },
+            ],
+            { useNativeDriver: false }
+          )(e, { dx: Math.min(INITIAL_OFFSET, initialPosition + gestureState.dx) })
+
+          Animated.event(
+            [
+              null,
+              {
+                moveX: backgroundAnim
+              },
+            ],
+            { useNativeDriver: false }
+          )(e, { moveX: Math.min(gestureState.moveX / width, MAX_BG_OPACITY) })
+        }
+      }
+    },
+    onPanResponderRelease: (e, gestureState) => {
+      if (isOpen) {
+        // Close if open & drag is past cutoff
+        if (gestureState.vx < 0 && gestureState.moveX < MOVE_CUTOFF_CLOSE * width) {
+          slideOut()
+        } else {
+          slideIn()
+        }
+      } else {
+        // Open if closed & drag is past cutoff
+        if (gestureState.vx > 0 && gestureState.moveX > MOVE_CUTOFF_OPEN * width) {
+          slideIn()
+          open()
+        } else {
+          slideOut()
+        }
+      }
+    },
+  })
+
+  useEffect(() => {
+    if (isOpen) {
+      slideIn()
+    }
+  }, [isOpen, slideIn])
+
+  const onGoToRoute = useCallback((route: string) => {
+    if (webRef.current) {
+      postMessage(webRef.current, {
+        type: MessageType.PUSH_ROUTE,
+        route,
+        isAction: true
+      })
+    }
+    slideOut()
+  }, [webRef, slideOut])
+
+  const containerStyle = useTheme(styles.container, {
+    backgroundColor: 'background'
+  })  
+
+  return (
+    <>
+      {/* animated drawer */}
+      <Animated.View
+        {...panResponder.panHandlers}
+        style={[
+          styles.animator,
+          {
+            transform: [
+              {
+                translateX: translationAnim
+              }
+            ]
+          }
+        ]}
+      >
+        <View style={containerStyle}>
+          <TopBar onClose={slideOut} />
+          <List
+            onLoadMore={onLoadMore}
+            onRefresh={onRefresh}
+            onGoToRoute={onGoToRoute}
+          />
+        </View>
+      </Animated.View>
+
+      {/* background overlay */}
+      <Animated.View
+        pointerEvents='none'
+        style={[
+          styles.background,
+          {
+            opacity: backgroundAnim,
+          }
+        ]}
+      />
+    </>
+  )
+}
+
+const mapStateToProps = (state: AppState) => ({
+  isOpen: getIsOpen(state)
+})
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  open: () => dispatch(notificationsActions.open()),
+  close: () => dispatch(notificationsActions.close()),
+  markAsViewed: () => dispatch(notificationsActions.markAsViewed())
+})
+
+export default connect(mapStateToProps, mapDispatchToProps)(Notifications)
```

Comment:
Could be using `useSelector`, `useDispatch` if you wanted to 🙃  - no need to change.